### PR TITLE
Make pages glob pattern overridable

### LINKED ISSUE #3183: The build assumes that all js files under /pages is a page
We have structured our files so that page specific code is in the pages folder. Here is an example:

- pages
  - manage-users
    - \_\_tests\_\_
      - manage-users.test.js
    - index.js
    - components
      - ManageUsersPage.js
      - \<other page specific components here\>

The index.js file has the ManageUsersPage.js components as its default export. That is a regular next.js page component

This structure has worked fine for us except when we build. This hardcoded glob assumes that all js files under pages is a page: https://github.com/zeit/next.js/blob/canary/server/build/webpack.js#L50

Does it make sense to make this glob overridable or do we have to not use file based routing?

### PR DESCRIPTION
**Pull request by emattias**
This fixes #3183

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]
Don't see any harm in adding this 👍 I do want to keep this as 'internal' (undocumented) cause it's not a practice we endorse.

**Comment by zenflow** [comment]
@timneutkens Gotta ask, why is it not an endorsed practice? 

(Might be better to have a statement on that before we link this PR up with all the duplicates of issue #3183 ... Theres at least 4)

File: server/build/webpack.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export default async function createCompiler (dir, { buildId, dev = false, quiet
       ]
     }
 
-    const pages = await glob('pages/**/*.js', { cwd: dir })
+    const pages = await glob(config.pagesGlobPattern, { cwd: dir })
     const devPages = pages.filter((p) => p === 'pages/_document.js' || p === 'pages/_error.js')
```
Comment (alexindigo) [inline server/build/webpack.js]: This would stil "lock" override pages within "pages" directory?

**Comment by djskinner** [comment]
Why not put this in the docs? Even if it's their with a caveat. My test files are colocated with the implementations . I don't want to have to do something different just for the pages.

**Comment by timneutkens** [comment]
It's not in the docs as this was removed.

**Comment by jpellizzari** [comment]
> It's not in the docs as this was removed.

As in this PR was reverted? Can we get a link to the revert?

**Comment by zenflow** [comment]
@jpellizzari someone could probably find where this change was reverted.. but what would be the point?

**Comment by probablyup** [comment]
It would be great if next automatically ignored files with an underscore prefix inside the pages folder. That way relevant files can be nicely co-located.

**Pull request merged**